### PR TITLE
Scrape reddit

### PR DESCRIPTION
**Pull request by nod0n**
will me merge for now and documentation is still todo

TODO: check TODO in code
TODO: more then 100 posts
TODO: Comments
TODO: more coins and more subreddits
TODO: scrape function needs parameter: headless has to be possible -> slurm & github
TODO: document modules, public variables, classes, class variables, functions, function parameters

### CODE REVIEW

**Comment by gabrieltorresgamez** [comment]
@FloBaumgartner should also review this when the PR is ready, since it's your project and i'm only interested on the functionality. But i can advise you to install these three vscode extensions:
- Pylint - usual linting for code
- Mypy - advanced type checking, might need to pip install the mypy module on your python env
- sourcery - more advanced code recommendations for cleaner code

this will improve your code drastically and correct typing will keep you from unknowingly accessing None values which might throw errors

File: src/CryptoFraudDetection/scraper/reddit.py [hunk 1]
```diff
@@ -5,5 +5,147 @@
...
+import random
+from selenium import webdriver
+from selenium.common.exceptions import TimeoutException, NoSuchElementException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+import time
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
```
Comment (gabrieltorresgamez) [inline src/CryptoFraudDetection/scraper/reddit.py:23]: pass logger into RedditScraper init

File: src/CryptoFraudDetection/scraper/reddit.py [hunk 2]
```diff
@@ -5,5 +5,147 @@
...
+            return element
+        except TimeoutException:
+            logger.error(f"Timeout waiting for element with locator: {locator}")
+            return None
+
+
+    def search_posts(self) -> None:
+        """Search for posts in a specific subreddit."""
+        url = f"{self._base_url}/{self._subreddit}/search?q={self._search_query}&restrict_sr=on&sort=new&t=all&limit={self._limit}&after=id___"
+        self.driver.get(url)   # TODO: error handling (differentiate between exceptions and maybe try again? sometimes dying might be ok)
```
Comment (gabrieltorresgamez) [inline src/CryptoFraudDetection/scraper/reddit.py:69]: yes, use specific catching

**Comment by gabrieltorresgamez** [comment]
i would also advice you to get some inspiration from current scrapers in this repo, for example return a dict instead of a dataframe, so it's consistent across the repo

**Review by FloBaumgartner — APPROVED** [review]

**Review by FloBaumgartner — APPROVED** [review]

**Pull request merged**